pkg/syncer/statussyncer.go

```diff
@@ -28,24 +29,29 @@ import (
 	"k8s.io/client-go/dynamic"
 	"k8s.io/client-go/rest"
 	"k8s.io/klog/v2"
+
+	workloadv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/workload/v1alpha1"
 )
 
-func deepEqualStatus(oldObj, newObj interface{}) bool {
+func deepEqualFinalizersAndStatus(oldObj, newObj interface{}) bool {
```

Comment:
simplify this by typing oldObj and newObj as unstructured. Where this is called we know everything is unstructured. That will also remove some test cases in the unit tests.